src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs

```diff
@@ -90,9 +120,85 @@ public async Task ProcessEventsAsync(IEnumerable<IEventMessage> events, string t
                 }
             }
 
-            // Step 3: Transform the Measurement Groups into Observations and write to the FHIR Server.
-            //         Exceptions are currently handled in ProcessMeasurementGroups
-            await ProcessMeasurementGroups(measurementGroups, template, log, measurementToEventMapping).ConfigureAwait(false);
+            // Transform the Measurement Groups into Observations and write to the FHIR Server.
+            // Exceptions are currently handled in SendMeasurementGroups
+            await SendMeasurementGroups(measurementGroups, template, log, measurementToEventMapping).ConfigureAwait(false);
+        }
+
+        private async Task ProcessMeasurementGroupEvents(IEnumerable<IEventMessage> events, ILookupTemplate<IFhirTemplate> template, ITelemetryLogger log)
+        {
+            if (events.Count() == 0)
+            {
+                return;
+            }
+
+            foreach (var evt in events)
+            {
+                try
+                {
+                    IEnumerable<MeasurementGroup> groupedMeasurements;
+
+                    try
+                    {
+                        var partitionId = events.FirstOrDefault()?.PartitionId;
+
+                        // decompress the measurement group if it is compressed
+                        byte[] bytes;
+                        if (evt.BodyContentType == "application/gzip")
+                        {
+                            bytes = Compression.DecompressWithGzip(evt.Body.ToArray());
+                            evt.Body = new ReadOnlyMemory<byte>(bytes);
+                        }
+                        else

```

Comment:
Personally, I prefer using the switch expressions for scenarios like this.  See https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression.  It doesn't really matter for one if/else statement but if we add other types the switch expression I feel results in more readable code compared to a large if/if else/else statement.

It would also be good to have a scenario if content is set to a type we don't support/know and raise an error